modules/multiplayer/multiplayer_synchronizer.h

```diff
@@ -46,12 +46,20 @@ class MultiplayerSynchronizer : public Node {
 	};
 
 private:
+	struct Watcher {
+		NodePath prop;
+		uint64_t last_change_msec = 0;
```

Comment:
I wonder if this should be stored as microseconds, so that framerates higher than 1,000 don't break this design entirely (which is entirely possible on a headless server with no rendering, at least in the initial stages of a project).

Reply:
I'll look into it, but having a framerate higher than 1000 with the default options (i.e. sync on every frame) would likely cause weird issues in any case (as you'll end up consuming MiB/s per client).
Maybe we can add a "max network tick" settings option to a reasonable default to avoid this? (i.e. the replication tick will not go above that even if the framerate is higher).

Reply:
Don't we already have a replication interval setting? Does synchronizing on change only bypass this replication interval (which makes sense to reduce latency)?

In this case, I guess we could have a property and project setting to define a hard cap on how often properties can be synchronized.

Reply:
I've added a new `delta_interval` property to control how often the watched properties are checked for changes.

I think both the project settings caps and custom functions for detecting changes are good ideas, but should go in a separate PR since this one has grown big enough :sweat_smile: